java-frontend/src/test/files/se/ConditionAlwaysTrueOrFalseCheck.java

```diff
@@ -1703,6 +1734,38 @@ void foo() {
       if (volatileField) {} // Compliant as this field is volatile, it can be modified by another thread
     }
   }
+class DITO {
+  DITO() throws MyExceptionFoo {}
+}
+  class MyExceptionFoo {}
+  void plop(Object result) {
+    if (result == null) {
+      // not found, try to create it
+      try {
+        result = new DITO();
+      } catch (final MyExceptionFoo ie) {
+                /* Swallow IntrospectionException
+                 * TODO: Why?
+                 */
+      }
+      if (result != null) {
```

Comment:
I would add a comment to explicitly saying that this condition is compliant, and clean the code snippet to only contain relevant part (useless comments, call to unknown methods from unknown object, unknown variable `name`)